crates/exex/exex/src/backfill.rs

```diff
@@ -0,0 +1,306 @@
+use reth_evm::execute::{BatchExecutor, BlockExecutionError, BlockExecutorProvider};
+use reth_exex_types::BackfillThresholds;
+use reth_node_api::FullNodeComponents;
+use reth_node_core::node_config::NodeConfig;
+use reth_primitives::BlockNumber;
+use reth_provider::{
+    BlockReader, Chain, DatabaseProviderFactory, HeaderProvider, HistoricalStateProviderRef,
+    ProviderError, StaticFileProviderFactory, TransactionVariant,
+};
+use reth_prune_types::PruneModes;
+use reth_revm::database::StateProviderDatabase;
+use reth_stages_api::format_gas_throughput;
+use reth_tracing::tracing::{debug, trace};
+use std::{
+    ops::RangeInclusive,
+    time::{Duration, Instant},
+};
+
+/// Factory for creating new backfill jobs.
+#[derive(Debug, Clone)]
+pub struct BackfillJobFactory<Node: FullNodeComponents> {
+    components: Node,
+    config: NodeConfig,
+    thresholds: BackfillThresholds,
+}
+
+impl<Node: FullNodeComponents> BackfillJobFactory<Node> {
+    /// Creates a new backfill job factory.
+    pub const fn new(components: Node, config: NodeConfig, thresholds: BackfillThresholds) -> Self {
+        Self { components, config, thresholds }
+    }
+
+    /// Creates a new backfill job for the given range.
+    pub fn backfill(&self, range: RangeInclusive<BlockNumber>) -> BackfillJob<Node> {
+        BackfillJob {
+            components: self.components.clone(),
+            prune_modes: self.config.prune_config().unwrap_or_default().segments,
+            range,
+            thresholds: self.thresholds.clone(),
+        }
+    }
+}
+
+/// Backfill job started for a specific range.
+///
+/// It implements [`Iterator`] that executes blocks in batches and yields
+/// [notifications](`crate::ExExNotification)s.
```

Comment:
needs update, yields chain.

what's the batch?

Reply:
explained the batch